MerlinAU.asp

```diff
@@ -337,38 +340,170 @@ function ToggleDaysOfWeek (isEveryXDayChecked, numberOfDays)
+  for (var m=0; m<60; m+=minInc) selM.add(new Option(pad(m), m));
+
+  input.readOnly = true;
+  input.dataset.mauTimeApplied = '1';
+
+  function emit(el, type){
+    el.dispatchEvent(new Event(type, { bubbles:true }));
+  }
+
+  function apply(){
+    var hh = pad(+selH.value), mm = pad(+selM.value);
+    input.value = hh + ':' + mm;
+    if (window.ClearTimePickerInvalid) window.ClearTimePickerInvalid(input);
+    if (window.ValidateTimePicker) window.ValidateTimePicker(input);
+    emit(input,'input'); emit(input,'change');
+  }
+
+  if (input.nextSibling) input.parentNode.insertBefore(wrap, input.nextSibling);
+  else input.parentNode.appendChild(wrap);
+  wrap.appendChild(selH);
+  wrap.appendChild(document.createTextNode(' : '));
+  wrap.appendChild(selM);
+
+  input.setAttribute('aria-hidden','true');
+  input.tabIndex = -1;
+  selH.setAttribute('aria-label','Hour');
+  selM.setAttribute('aria-label','Minute');
+
+  var start = parseHHMM(input.value);
+  selH.value = start.h; selM.value = start.m;
+  selH.onchange = apply; selM.onchange = apply;
+  var flaggedInvalid = (input.getAttribute('aria-invalid') === 'true') ||
+                       (typeof fwTimeInvalidFromConfig !== 'undefined' && fwTimeInvalidFromConfig) ||
+                       (input.classList && input.classList.contains('Invalid'));
+
+  if (!flaggedInvalid) {
+    apply(); // normalize displayed HH:MM
+  }
+  
+  function syncDisabled(){
+    var dis = !!input.disabled;
+    selH.disabled = dis; selM.disabled = dis;
+    wrap.style.opacity = dis ? '0.5' : '1';
+  }
+  syncDisabled();
+
+  var mo = new MutationObserver(syncDisabled);
+  mo.observe(input, { attributes:true, attributeFilter:['disabled'] });
+
+  function syncFromInput(){
+    var v = (input.value || '').trim();
+    if (!v) return; // keep current selects if input was cleared for invalid state
+    var t = parseHHMM(v);
+    selH.value = t.h; selM.value = t.m;
+  }
+  input.addEventListener('input', syncFromInput);
+  input.addEventListener('change', syncFromInput);
 }
 
-/**-------------------------------------**/
-/** Added by Martinski W. [2025-Jan-24] **/
-/**-------------------------------------**/
-function ValidateScheduleMINS (theMINSstr)
-{
-   if (theMINSstr === null ||
-       theMINSstr.length == 0 ||
-       theMINSstr.length >= 3 ||
-       theMINSstr.match (`${numberRegExp}`) === null)
-   { return false; }
-   let theMINSNum = parseInt(theMINSstr, 10);
-   if (theMINSNum < 0 || theMINSNum > 59)
-   { return false; }
-   else
-   { return true; }
+/**---------------------------------------**/
+/** Added by ExtremeFiretop [2025-Aug-24] **/
+/**---------------------------------------**/
+function ParseTimeHHMM(v){
+  v = (v || '').trim();
+  if (!/^[0-2]\d:[0-5]\d$/.test(v)) return { ok:false };
+  var h = parseInt(v.slice(0,2),10), m = parseInt(v.slice(3,5),10);
+  if (h < 0 || h > 23 || m < 0 || m > 59) return { ok:false };
+  return { ok:true, h:h, m:m };
+}
```

Comment:
Same here: using one-letter variable names.
I'll stop here, but there are too many instances to ignore this bad coding practice.